contracts/multisig/src/types.rs

```diff
@@ -33,42 +32,10 @@ pub enum MultisigState {
     },
 }
 
-const MESSAGE_HASH_LEN: usize = 32;
 
-impl TryFrom<HexBinary> for MsgToSign {
-    type Error = ContractError;
-
-    fn try_from(other: HexBinary) -> Result<Self, Self::Error> {
-        if other.len() != MESSAGE_HASH_LEN {
-            return Err(ContractError::InvalidMessageFormat {
-                reason: "Invalid input length".into(),
-            });
-        }
-
-        Ok(MsgToSign(other))
+impl From<HexBinary> for MsgToSign {
+    fn from(value: HexBinary) -> Self {
+        MsgToSign(value)
```

Comment:
if I understand correctly the msg is now arbitrary length, but ampd is still signing on 32 bytes, right? in that case wouldn't you need to change how the verification works as well since now it will pass the whole message to the verification function? https://github.com/axelarnetwork/axelar-amplifier/blob/ba3d3c5117b2a89f527ff9c84f588ef9097ac51b/contracts/multisig/src/signing.rs#L85

When this rolls out, what would happen to messages that are currently being signed/verified?

Reply:
hmm this is a good point, the evm chains will still just pass 32 byte message hashes to be signed and it will work as is, and I would only think XRPL would pass larger messages. However, maybe we should only allow greater than 32 bytes when a sig verifier callback is attached